diff --git a/Makefile b/Makefile
index 49b1f6f..b5ced4f 100644
--- a/Makefile
+++ b/Makefile
@@ -66,7 +66,7 @@ endif
 ifdef CONFIG_CLANG
   HOST_CC=clang
   CC=$(CROSS_PREFIX)clang
-  CFLAGS=-g -Wall -MMD -MF $(OBJDIR)/$(@F).d
+  CFLAGS+=-g -Wall -MMD -MF $(OBJDIR)/$(@F).d
   CFLAGS += -Wextra
   CFLAGS += -Wno-sign-compare
   CFLAGS += -Wno-missing-field-initializers
@@ -87,7 +87,7 @@ ifdef CONFIG_CLANG
 else
   HOST_CC=gcc
   CC=$(CROSS_PREFIX)gcc
-  CFLAGS=-g -Wall -MMD -MF $(OBJDIR)/$(@F).d
+  CFLAGS+=-g -Wall -MMD -MF $(OBJDIR)/$(@F).d
   CFLAGS += -Wno-array-bounds -Wno-format-truncation
   ifdef CONFIG_LTO
     AR=$(CROSS_PREFIX)gcc-ar
diff --git a/quickjs.c b/quickjs.c
index 48aeffc..920c4e8 100644
--- a/quickjs.c
+++ b/quickjs.c
@@ -33577,6 +33577,8 @@ static JSValue __JS_EvalInternal(JSContext *ctx, JSValueConst this_obj,
                                  const char *input, size_t input_len,
                                  const char *filename, int flags, int scope_idx)
 {
+    extern size_t __afl_correctness_step;
+    __afl_correctness_step = 1;
     JSParseState s1, *s = &s1;
     int err, js_mode, eval_type;
     JSValue fun_obj, ret_val;
@@ -33587,6 +33589,7 @@ static JSValue __JS_EvalInternal(JSContext *ctx, JSValueConst this_obj,
     JSModuleDef *m;
 
     js_parse_init(ctx, s, input, input_len, filename);
+    __afl_correctness_step = 2;
     skip_shebang(s);
 
     eval_type = flags & JS_EVAL_TYPE_MASK;
@@ -33596,8 +33599,11 @@ static JSValue __JS_EvalInternal(JSContext *ctx, JSValueConst this_obj,
         sf = ctx->rt->current_stack_frame;
         assert(sf != NULL);
         assert(JS_VALUE_GET_TAG(sf->cur_func) == JS_TAG_OBJECT);
+        __afl_correctness_step = 3;
         p = JS_VALUE_GET_OBJ(sf->cur_func);
+        __afl_correctness_step = 4;
         assert(js_class_has_bytecode(p->class_id));
+        __afl_correctness_step = 5;
         b = p->u.func.function_bytecode;
         var_refs = p->u.func.var_refs;
         js_mode = b->js_mode;
@@ -33612,17 +33618,27 @@ static JSValue __JS_EvalInternal(JSContext *ctx, JSValueConst this_obj,
             js_mode |= JS_MODE_STRIP;
         if (eval_type == JS_EVAL_TYPE_MODULE) {
             JSAtom module_name = JS_NewAtom(ctx, filename);
-            if (module_name == JS_ATOM_NULL)
+            __afl_correctness_step = 3;
+            if (module_name == JS_ATOM_NULL){
+                __afl_correctness_step = 4;
                 return JS_EXCEPTION;
+            }
             m = js_new_module_def(ctx, module_name);
-            if (!m)
+            __afl_correctness_step = 5;
+            if (!m){
+                __afl_correctness_step = 6;
                 return JS_EXCEPTION;
+            }
             js_mode |= JS_MODE_STRICT;
         }
     }
+    __afl_correctness_step = 7;
     fd = js_new_function_def(ctx, NULL, TRUE, FALSE, filename, 1);
-    if (!fd)
+    __afl_correctness_step = 8;
+    if (!fd){
+        __afl_correctness_step = 9;
         goto fail1;
+    }
     s->cur_func = fd;
     fd->eval_type = eval_type;
     fd->has_this_binding = (eval_type != JS_EVAL_TYPE_DIRECT);
@@ -33640,40 +33656,61 @@ static JSValue __JS_EvalInternal(JSContext *ctx, JSValueConst this_obj,
     }
     fd->js_mode = js_mode;
     fd->func_name = JS_DupAtom(ctx, JS_ATOM__eval_);
+    __afl_correctness_step = 11;
     if (b) {
-        if (add_closure_variables(ctx, fd, b, scope_idx))
+        __afl_correctness_step = 12;
+        if (add_closure_variables(ctx, fd, b, scope_idx)) {
+            __afl_correctness_step = 13;
             goto fail;
+        }
     }
     fd->module = m;
     s->is_module = (m != NULL);
     s->allow_html_comments = !s->is_module;
+    __afl_correctness_step = 14;
 
     push_scope(s); /* body scope */
+    __afl_correctness_step = 15;
     fd->body_scope = fd->scope_level;
     
     err = js_parse_program(s);
+    __afl_correctness_step = 16;
     if (err) {
     fail:
+        __afl_correctness_step = 17;
         free_token(s, &s->token);
         js_free_function_def(ctx, fd);
+        __afl_correctness_step = 18;
         goto fail1;
     }
 
     /* create the function object and all the enclosed functions */
     fun_obj = js_create_function(ctx, fd);
-    if (JS_IsException(fun_obj))
+    __afl_correctness_step = 19;
+    if (JS_IsException(fun_obj)) {
+        __afl_correctness_step = 20;
         goto fail1;
+    }
     /* Could add a flag to avoid resolution if necessary */
     if (m) {
         m->func_obj = fun_obj;
-        if (js_resolve_module(ctx, m) < 0)
+        if (js_resolve_module(ctx, m) < 0) {
+            __afl_correctness_step = 21;
             goto fail1;
+        }
         fun_obj = JS_DupValue(ctx, JS_MKPTR(JS_TAG_MODULE, m));
+        __afl_correctness_step = 22;
     }
     if (flags & JS_EVAL_FLAG_COMPILE_ONLY) {
         ret_val = fun_obj;
     } else {
         ret_val = JS_EvalFunctionInternal(ctx, fun_obj, this_obj, var_refs, sf);
+        __afl_correctness_step = 23;
+        if (JS_IsException(ret_val)) {
+            __afl_correctness_step = 24;
+        } else {
+            __afl_correctness_step = 25;
+        }
     }
     return ret_val;
  fail1:
